### ISSUE TITLE
new model

### ISSUE THREAD

**Issue by shunxing12345** [SafeAILab/EAGLE#66]
Hi i want to add a model that has a different architecture from the LLaMA model. BUT when I was trying
`accelerate launch -m --mixed_precision=bf16 eagle.train.main --tmpdir [path of data]\
--cpdir [path of checkpoints] -- configpath [path of config file]`
I got the following ERROR
![image](https://github.com/SafeAILab/EAGLE/assets/168084185/3940ff1b-c40f-4bbb-91b5-a460ace8b7be)

**Comment by Liyuhui-12** [comment]
It seems that the name of the embedding in your model is not 'embed_tokens'. You can modify it to the name of the embedding layer in your model.

**Comment by shunxing12345** [comment]
thanks for your replay!
I have modified this parameter, but I got a new problem
![image](https://github.com/SafeAILab/EAGLE/assets/168084185/a721b302-e296-4623-b216-ae5eaed56e47)
it is a 12B model
A100 40g * 8

**Comment by shunxing12345** [comment]
Thanks for your replay!
I got an other problem I am trying to train an LLM which structure differs from LLaMA and Mixtral,　should I change the code of cnet.py? It seems based on LLaMA

**Comment by Liyuhui-12** [comment]
This is not necessary; EAGLE's structure is independent of the target model. You can use the same cnet.py, or you can try other structures as well.

**Comment by shunxing12345** [comment]
Thanks!
I have a finetuned a 12B model, but I got the OOM ERROR in `model, head, optimizer, train_loader, test_loader, scheduler = accelerator.prepare(
        model, head, optimizer, train_loader, test_loader, scheduler`. I have 8 40G-A100.
![image](https://github.com/SafeAILab/EAGLE/assets/168084185/098c15ee-18a6-4d54-8ca7-aa97e4b7cf7e)
this is my train_config
![image](https://github.com/SafeAILab/EAGLE/assets/168084185/e414f526-23a0-4cc8-b873-78226e263420)
this is my config.json
![image](https://github.com/SafeAILab/EAGLE/assets/168084185/92de2bc0-a14e-417f-9d8c-32e054cd52d1)

**Comment by Liyuhui-12** [comment]
I noticed that your "n_layers" is set to 38, which makes your draft model very large. In EAGLE, the draft model consists of only one layer.

**Comment by shunxing12345** [comment]
Hi, I have successfully trained an Auto-regression Head, but I encountered the following error during inference.
https://github.com/SafeAILab/EAGLE/blob/main/eagle/modeling_eagle.py#L957
![image](https://github.com/SafeAILab/EAGLE/assets/168084185/8817c8ed-a093-4699-8fb0-19adb7a109f0)
and here is the size of Tensor
![image](https://github.com/SafeAILab/EAGLE/assets/168084185/fc905f43-7f6f-42b7-b63a-861d0a5a61b6)
